src/test/java/org/javarosa/xform/parse/XFormParserTest.java

```diff
@@ -130,6 +132,19 @@ public void parsesPreloadForm() throws IOException {
         assertEquals("Sample Form - Preloading", formDef.getTitle());
     }
 
+    @Test
+    public void parsesExternalSecondaryInstanceAsFormInstance() throws IOException {
+        Path form = r("Sample-Preloading.xml");
+        setUpSimpleReferenceManager("file-csv", form.getParent());
+        FormDef formDef = parse(form);
+        Enumeration<DataInstance> elements = formDef.getNonMainInstances();
+
+        while (elements.hasMoreElements()) {
+            DataInstance instance = elements.nextElement();
+            assertTrue(instance instanceof FormInstance);
```

Comment:
This looks like the opposite of what is desired but perhaps some comments would help clarify. In that form, an instance is declared with `<instance id="hhplotdetails" src="jr://file-csv/hhplotdetails.csv">`. However, it's never referred to as `instance('hhplotdetails')` so the file should not be loaded as an in-memory instance.